src/Raven.Server/Documents/Sharding/Executors/AllOrchestratorNodesExecutor.cs

```diff
@@ -53,59 +50,29 @@ public RequestExecutor GetRequestExecutorForNode(string tag)
         return executor;
     }
 
-    private void RegisterForTopologyChange()
+    private void UpdateExecutors(ClusterTopology clusterTopology, OrchestratorTopology orchestrator)
     {
-        var task = _store.Engine.GetTopologyChanged();
-        task.ContinueWith(_ =>
-        {
-            RegisterForTopologyChange();
-
-            var topology = _store.GetClusterTopology();
-            Interlocked.Exchange(ref _clusterTopology, topology);
-
-            UpdateExecutors(topology, _record.Sharding.Orchestrator.Topology);
-        }, TaskContinuationOptions.OnlyOnRanToCompletion);
-    }
-
-    private void UpdateExecutors(ClusterTopology cluster, OrchestratorTopology orchestrator)
-    {
-        var disposables = new List<IDisposable>();
-
         var oldCurrent = _current;
         var newCurrent = new Dictionary<string, RequestExecutor>(StringComparer.OrdinalIgnoreCase);
 
-        foreach (var node in cluster.AllNodes)
+        PublishedUrls published;
+        using (ServerStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+        using (context.OpenReadTransaction())
+        {
+            published = PublishedUrls.Read(context);
+        }
+
+        foreach (var node in clusterTopology.AllNodes)
         {
             var tag = node.Key;
-            var url = node.Value;
 
             if (orchestrator.AllNodes.Contains(tag, StringComparer.OrdinalIgnoreCase) == false)
                 continue;
 
-            if (oldCurrent.TryGetValue(tag, out var requestExecutor) == false)
-            {
-                newCurrent[tag] = RequestExecutor.CreateForSingleNodeWithoutConfigurationUpdates(url, _record.DatabaseName, _store.Server.Certificate.Certificate,
-                    DocumentConventions.DefaultForServer);
-                continue;
-            }
-
-            if (string.Equals(requestExecutor.Url, url, StringComparison.OrdinalIgnoreCase) == false)
-            {
-                disposables.Add(requestExecutor); // will dispose outside the lock
-                newCurrent[tag] = RequestExecutor.CreateForSingleNodeWithoutConfigurationUpdates(url, _record.DatabaseName, _store.Server.Certificate.Certificate,
-                    DocumentConventions.DefaultForServer);
-                continue;
-            }
-
-            if (_store.Server.Certificate.Certificate?.Thumbprint != requestExecutor.Certificate?.Thumbprint)
-            {
-                disposables.Add(requestExecutor);
-                newCurrent[tag] = RequestExecutor.CreateForSingleNodeWithoutConfigurationUpdates(url, _record.DatabaseName, _store.Server.Certificate.Certificate,
-                    DocumentConventions.DefaultForServer);
-                continue;
-            }
+            var url = published.SelectUrl(tag, clusterTopology);
```

Comment:
you removed all the code that was checking if we need to update the executor at all incl. certificate changes, etc?

Reply:
we will update all request executor on certificate change